src/view-resources.js

```diff
@@ -67,6 +73,72 @@ interface ViewEngineHooks {
 * Represents a collection of resources used during the compilation of a view.
 */
 export class ViewResources {
+
+  static convention(target: Function) {
+    let config = target.resource;
+    let resource;
+    if (config) {
+      if (typeof config === 'string') {
+        // it's a custom element, with name is the resource variable
+        // static resource = 'my-element'
+        resource = new HtmlBehaviorResource();
+        resource.elementName = config;
+      } else {
+        if (typeof config === 'function') {
+          // static resource() {  }
+          config = config();
+        }
+        if (typeof config === 'string') {
+          // static resource() { return 'my-custom-element-name' }
+          config = { name: config };
+        }
+        // after normalization, copy to another obj
+        config = Object.assign({}, config);
+        // no type specified = custom element
+        let resourceType = config.type || 'element';
+        switch (resourceType) {
+          case 'element': case 'attribute':
+            resource = new HtmlBehaviorResource();
+            if (resourceType === 'element') {
+              resource.elementName = _hyphenate(config.name || target.name);
+            } else {
+              resource.attributeName = _hyphenate(config.name || target.name);
+            }
+            if (config.templateController) {
+              config.liftsContent = config.templateController;
+              delete config.templateController;
+            }
+            // one time default binding mode is 0, cannot check for falsy value ... T_T
+            if ('defaultBindingMode' in config) {
+              config.attributeDefaultBindingMode = config.defaultBindingMode;
+              delete config.defaultBindingMode;
+            }
+            // just copy over. Devs are responsible for what they specify in the config. good luck!.
+            Object.assign(resource, config);
+            break;
+          case 'valueConverter':
+            resource = new ValueConverterResource(camelCase(config.name || target.name)); break;
+          case 'bindingBehavior':
+            resource = new BindingBehaviorResource(camelCase(config.name || target.name)); break;
+          case 'viewEngineHooks':
+            resource = new ViewEngineHooksResource(); break;
+        }
+      }
+      // check for bindable registration
+      let bindables = typeof config === 'string' ? undefined : config.bindables;
+      if (Array.isArray(bindables) && resource instanceof HtmlBehaviorResource) {
+        for (let i = 0, ii = bindables.length; ii > i; ++i) {
+          let prop = bindables[i];
+          if (!prop || (typeof prop !== 'string' && !prop.name)) {
+            throw new Error('Invalid bindable property. Expected either a string or an object with "name" property.');
```

Comment:
It would be very helpful for users to have context in errors.
For instance the name of class being processed `target.name` (during dev it wouldn't be obfuscated / minified away) and maybe the index `i` of faulty bindable.

Reply:
I missed this. Will add a fix